### PR TITLE
test: support running integration test with envoy bin

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: api/plugins/tests/integration/dataplane/data_plane.go [hunk 1]
```diff
@@ -325,8 +393,13 @@ func (dp *DataPlane) Stop() {
...
 
 	logger.Info("stop envoy")
-	cmd := exec.Command("docker", "stop", containerName)
-	err = cmd.Run()
+
+	if isBinaryMode() {
+		dp.cmd.Process.Signal(syscall.SIGTERM)
+	} else {
+		cmd := exec.Command("docker", "stop", containerName)
+		err = cmd.Run()
```
Comment (zll600) [inline api/plugins/tests/integration/dataplane/data_plane.go:401]: I think that we can check `err` here immediately, another branch will not return an error.

**Comment by zll600** [comment]
If my points have mistakes, feel free to correct me.

File: api/plugins/tests/integration/dataplane/bootstrap.go [hunk 2]
```diff
@@ -150,15 +157,19 @@ func (b *bootstrap) buildConfiguration() (map[string]interface{}, error) {
...
 	clusters := staticResources["clusters"].([]interface{})
 
-	port := "9999"
-	portEnv := os.Getenv("TEST_ENVOY_CONTROL_PLANE_PORT")
-	if portEnv != "" {
+	port := 9999
+	portEnv, _ := strconv.Atoi(os.Getenv("TEST_ENVOY_CONTROL_PLANE_PORT"))
+	if portEnv != 0 {
 		port = portEnv
 	}
```
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/bootstrap.go:164]: No, thanks! My code style is more concise.

File: api/plugins/tests/integration/dataplane/data_plane.go [hunk 3]
```diff
@@ -246,24 +275,49 @@ func StartDataPlane(t *testing.T, opt *Option) (*DataPlane, error) {
...
-	if err != nil {
-		return nil, err
+	if isBinaryMode() {
+		// Like the standard mode, we use stdout file to store the log of Envoy
+		stdout, err := os.Create(filepath.Join(dir, "stdout"))
+		if err != nil {
+			return nil, err
+		}
+		cmd.Stdout = stdout
+
```
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:299]: I prefer to keep the code in the if-else block so I can write comments for it, especially to make the comment match the one in the stderr.

File: api/plugins/tests/integration/dataplane/data_plane.go [hunk 4]
```diff
@@ -209,19 +152,99 @@ func StartDataPlane(t *testing.T, opt *Option) (*DataPlane, error) {
...
+	cfgFile, err := os.Create(filepath.Join(dir, cfgFilename))
+	if err != nil {
+		return nil, err
+	}
+
+	err = opt.Bootstrap.WriteTo(cfgFile)
+	cfgFile.Close()
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:165]: We only use `cfgFile.Name()` later

File: api/plugins/tests/integration/dataplane/data_plane.go [hunk 5]
```diff
@@ -209,19 +152,99 @@ func StartDataPlane(t *testing.T, opt *Option) (*DataPlane, error) {
...
+			soPath + ":/etc/libgolang.so" +
+			" -v /tmp:/tmp" +
+			" -e GOCOVERDIR=" + coverDir +
+			" " + strings.Join(envs, " ") +
+			" -p " + dataPlanePort + ":" + dataPlanePort +
+			" -p " + adminAPIPort + ":" + adminAPIPort + " " +
+			hostAddr + " " +
+			image
+
+		envoyCmd = "envoy -c /etc/envoy.yaml"
```
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:243]: This is too trivial to change.

File: api/plugins/tests/integration/dataplane/data_plane.go (hunk 1 above)
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:401]: This is too trivial to change. If we add the err next time, we have to move the err check out again.

File: api/plugins/tests/integration/dataplane/data_plane.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:165]: In fact, we must close the cfgFile immediately to ensure the written content is flushed.

File: api/plugins/tests/integration/dataplane/data_plane.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (spacewander) [reply api/plugins/tests/integration/dataplane/data_plane.go:243]: The cmdLine doesn't depend on the envoyCmd. And envoyCmd is a new neighbor of the cmdLine. Add the new code after the old one is natural.

File: api/plugins/tests/integration/dataplane/data_plane.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (zll600) [reply api/plugins/tests/integration/dataplane/data_plane.go:165]: OK, Thank you for your detailed explanation.

File: api/plugins/tests/integration/dataplane/data_plane.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (zll600) [reply api/plugins/tests/integration/dataplane/data_plane.go:243]: OK, it makes sense to me.

**Pull request merged**